From 9591ee225498341a49fa6fc5fc26d77d4e2b3f0d Mon Sep 17 00:00:00 2001
From: Peter Wu <lekensteyn@gmail.com>
Date: Tue, 6 Aug 2013 15:45:19 +0200
Subject: [PATCH 01/24] Fix wrong names for HID++ 1.0 error codes

---
 hidraw.c  | 26 +++++++++++++-------------
 ltunify.c | 26 +++++++++++++-------------
 2 files changed, 26 insertions(+), 26 deletions(-)

--- a/hidraw.c
+++ b/hidraw.c
@@ -82,19 +82,19 @@ static const char * report_types[0x100]
 
 static const char * error_messages[0x100] = {
 	// error messages for type=8F (ERROR_MSG)
-	[0x01] = "SUCCESS",
-	[0x02] = "INVALID_SUBID",
-	[0x03] = "INVALID_ADDRESS",
-	[0x04] = "INVALID_VALUE",
-	[0x05] = "CONNECT_FAIL",
-	[0x06] = "TOO_MANY_DEVICES",
-	[0x07] = "ALREADY_EXISTS",
-	[0x08] = "BUSY",
-	[0x09] = "UNKNOWN_DEVICE",
-	[0x0a] = "RESOURCE_ERROR",
-	[0x0b] = "REQUEST_UNAVAILABLE",
-	[0x0c] = "INVALID_PARAM_VALUE",
-	[0x0d] = "WRONG_PIN_CODE",
+	[0x00] = "SUCCESS",
+	[0x01] = "INVALID_SUBID",
+	[0x02] = "INVALID_ADDRESS",
+	[0x03] = "INVALID_VALUE",
+	[0x04] = "CONNECT_FAIL",
+	[0x05] = "TOO_MANY_DEVICES",
+	[0x06] = "ALREADY_EXISTS",
+	[0x07] = "BUSY",
+	[0x08] = "UNKNOWN_DEVICE",
+	[0x09] = "RESOURCE_ERROR",
+	[0x0A] = "REQUEST_UNAVAILABLE",
+	[0x0B] = "INVALID_PARAM_VALUE",
+	[0x0C] = "WRONG_PIN_CODE",
 };
 
 // I don't know the upper bound, perhaps 0x10 is enough
--- a/ltunify.c
+++ b/ltunify.c
@@ -225,19 +225,19 @@ struct receiver_info receiver;
 
 // error messages for type=8F (ERROR_MSG)
 static const char * error_messages[0x100] = {
-	[0x01] = "SUCCESS",
-	[0x02] = "INVALID_SUBID",
-	[0x03] = "INVALID_ADDRESS",
-	[0x04] = "INVALID_VALUE",
-	[0x05] = "CONNECT_FAIL",
-	[0x06] = "TOO_MANY_DEVICES",
-	[0x07] = "ALREADY_EXISTS",
-	[0x08] = "BUSY",
-	[0x09] = "UNKNOWN_DEVICE",
-	[0x0a] = "RESOURCE_ERROR",
-	[0x0b] = "REQUEST_UNAVAILABLE",
-	[0x0c] = "INVALID_PARAM_VALUE",
-	[0x0d] = "WRONG_PIN_CODE",
+	[0x00] = "SUCCESS",
+	[0x01] = "INVALID_SUBID",
+	[0x02] = "INVALID_ADDRESS",
+	[0x03] = "INVALID_VALUE",
+	[0x04] = "CONNECT_FAIL",
+	[0x05] = "TOO_MANY_DEVICES",
+	[0x06] = "ALREADY_EXISTS",
+	[0x07] = "BUSY",
+	[0x08] = "UNKNOWN_DEVICE",
+	[0x09] = "RESOURCE_ERROR",
+	[0x0A] = "REQUEST_UNAVAILABLE",
+	[0x0B] = "INVALID_PARAM_VALUE",
+	[0x0C] = "WRONG_PIN_CODE",
 };
 
 static const char * device_type[0x10] = {
