From 89dce898377abfdde54d784b050841a282ca3151 Mon Sep 17 00:00:00 2001
From: Peter Wu <lekensteyn@gmail.com>
Date: Sat, 28 Sep 2013 16:37:34 +0200
Subject: [PATCH 16/24] ltunify: fix crash on unrecognised parameter

---
 ltunify.c | 1 +
 1 file changed, 1 insertion(+)

--- a/ltunify.c
+++ b/ltunify.c
@@ -1067,6 +1067,7 @@ static int validate_args(int argc, char
 	struct option longopts[] = {
 		{ "device",     1, NULL, 'd' },
 		{ "help",       0, NULL, 'h' },
+		{ 0, 0, 0, 0 },
 	};
 
 	*argsp = NULL;
