From 1938dc08d1b0111f4eb236e4b920a321e78cb5da Mon Sep 17 00:00:00 2001
From: Peter Wu <lekensteyn@gmail.com>
Date: Wed, 19 Feb 2014 11:13:40 +0100
Subject: [PATCH 18/24] Warn when CONFIG_HIDRAW is disabled

---
 ltunify.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/ltunify.c
+++ b/ltunify.c
@@ -1208,6 +1208,10 @@ int open_hidraw(void) {
 				"for %s\n", hiddev_name);
 		} else {
 			fprintf(stderr, "No Logitech Unifying Receiver device found\n");
+			if (access("/sys/class/hidraw", R_OK)) {
+				fputs("The kernel must have CONFIG_HIDRAW enabled.\n",
+					stderr);
+			}
 			if (access("/sys/module/hid_logitech_dj", F_OK)) {
 				fprintf(stderr, "Driver is not loaded, try:"
 						"   sudo modprobe hid-logitech-dj\n");
